### DISCUSSION TITLE
Built-in Camera is not recognised.

### DISCUSSION THREAD

**Discussion by heikebotaru** [CobanRamo/Lenovo-IdeaPad-S340-15IIL#11] [Q&A]
Hello CobanRamo,
Thank you for your good job. I have tried the new EFI. The problem with the screen going dark on startup has been fixed and I think it works great. I have a problem with my Ideapad. The built-in camera is not recognised, so I can't use Zoom and Facetime. Is it a problem with the USBMap.kext? How is it with your Ideapad? I am using Monterey 12.2.

**Comment by CobanRamo** [comment]
The darker screen problem was solved with the WhateverGreen v1.5.7.
Camera is recognized and works normally for me.
You can't use my USBMap.kext of course, you have to do usb port mapping for your notebook.

**Reply by heikebotaru** [reply]
Thanks for the reply.
I will try to deal with it myself using Hackintool.app.

**Comment by CobanRamo** [comment]
![USB](https://user-images.githubusercontent.com/68192365/151621186-f283d714-da0e-41a4-b63b-171ec35e3c42.png)

**Comment by CobanRamo** [comment]
try this ssdt, am unsure if this fits your dsdt.
If it does not work send me your current DSDT and I will build you a SMBIOS and system independent SSDT.

[SSDT-USB-S340.aml.zip](https://github.com/CobanRamo/Lenovo-IdeaPad-S340-15IIL/files/7965247/SSDT-USB-S340.aml.zip)

**Reply by heikebotaru** [reply]
Hi, CobanRamo
Thanks lots.
I learned that there are several ways to deal with the limitation of the number of USB ports, such as
1. To apply patch to AppleUSBXHCIPCI.kext.
2. To set XhciPortLimit to true in config.plist.
3. To rewrite DSDT to specify the 15 USB ports to use.
4. To specify the 15 ports in USBInjectAll.kext and boot options.
5. To using USBInjectAll.kext and to create the SSDT that directs it to specify the 15 ports.
6. To create a kext file using Hackintool or USBMap.command.

I was able to successfully set up the USB Ports.
Method 6. worked well for me.
